service/controller.go

```diff
@@ -361,6 +374,44 @@ func (s *service) CreateVolume(
 			return nil, status.Errorf(codes.Unknown, "Create Volume %s failed with error: %v", volName, err)
 		}
 
+		// set quota limits, if specified in NFS storage class
+		isQuotaEnabled := s.opts.IsQuotaEnabled
+		if isQuotaEnabled {
+			// get filesystem (NFS volume), newly created
+			fs, err := system.GetFileSystemByIDName(fsResp.ID, "")
+			if err != nil {
+				Log.Debugf("Find Volume response: %v Error: %v", fs, err)
+			}
+			path, ok := params[KeyPath]
+			if !ok {
+				return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeyPath)
+			}
+
+			softLimit, ok := params[KeySoftLimit]
+			if !ok {
+				return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeySoftLimit)
+			}
+
+			gracePeriod, ok := params[KeyGracePeriod]
+			if !ok {
+				return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeyGracePeriod)
+			}
+
+			// create quota for the filesystem
+			if isQuotaEnabled {
```

Comment:
Why to check the isQuotaEnabled again here. I think we are checking this above at line no: 379

Reply:
updated